electron/main.js

```diff
@@ -0,0 +1,40 @@
+import { join } from "path";
+import { app, BrowserWindow, utilityProcess } from 'electron';
+
+const isAppPackaged = app.getVersion() !== '0.0.0-dev';
+const hollamaHost = isAppPackaged ? '0.0.0.0' : '127.0.0.1';
+const hollamaPort = isAppPackaged ? '4173' : '5173';
```

Comment:
I don't love this implementation but basically we assume we are in a `dev` environment if the version of `package.json` is `0.0.0-dev`, which is the default value.

This version get's overwritten by [`update-package-json.cjs`](https://github.com/fmaclen/hollama/blob/main/.github/workflows/publish/update-package-json.cjs) during the `publish.yml` GH action.

In `dev` mode we want to intentionally run Hollama the way we normally run it: `npm run dev`, which listens on `5173`.

When the _app is packaged_ (`production`), we bind the server to `0.0.0.0` so it can be accessed from other devices in the same LAN, and we use `4173` which is Vite's default "build" port.

Eventually we would probably need to allow the user to configure all of this stuff but it's fine for now.